controllers/admin/AdminAjaxPs_buybuttonliteController.php

```diff
@@ -0,0 +1,118 @@
+<?php
+/**
+* 2007-2016 PrestaShop
+*
+* DISCLAIMER
+*
+* Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+* versions in the future. If you wish to customize PrestaShop for your
+* needs please refer to http://www.prestashop.com for more information.
+*
+* @author    PrestaShop SA <contact@prestashop.com>
+* @copyright 2007-2015 PrestaShop SA
+* @license   http://addons.prestashop.com/en/content/12-terms-and-conditions-of-use
+* International Registered Trademark & Property of PrestaShop SA
+*/
+
+class AdminAjaxPs_buybuttonliteController extends ModuleAdminController
+{
+    public function ajaxProcessSearchProducts()
+    {
+        $context = Context::getContext();
+        $id_lang = $context->language->id;
+        $id_shop = $context->shop->id;
+        $query = pSQL(Tools::getValue('product_search'));
+
+        $sql = new DbQuery();
+        $sql->select('p.`id_product`, pa.`id_product_attribute`, pl.`name`, p.`reference`');
+        $sql->from('product', 'p');
+        $sql->join(Shop::addSqlAssociation('product', 'p'));
+        $sql->leftJoin('product_lang', 'pl', '
+            p.`id_product` = pl.`id_product`
+            AND pl.`id_lang` = ' . (int) $id_lang . Shop::addSqlRestrictionOnLang('pl')
+        );
+        $sql->leftJoin('product_attribute', 'pa', 'pa.`id_product` = p.`id_product`');
+        $where = 'pl.`name` LIKE \'%' . pSQL($query) . '%\'
+        OR p.`reference` LIKE \'%' . pSQL($query) . '%\'
+        OR EXISTS(SELECT * FROM `' . _DB_PREFIX_ . 'product_supplier` sp WHERE sp.`id_product` = p.`id_product` AND `product_supplier_reference` LIKE \'%' . pSQL($query) . '%\')';
```

Comment:
And this SQL query is complex. It obviously carries a lot of meaning and logic in it 😉 what about parse it in several parts easier to understand ?
Example:
```
$queryBeginning = '...';
/* (explanation about why this EXIST statement is required */
$firstExistStatement = '...':
/* (explanation about why this 2nd EXIST statement is required */
$secondExistStatement = '...';

$completeQuery = $queryBeginning . $firstExistStatement . $secondExistStatement . ...
```